### PR TITLE
🐛 Source Intercom: handle `conversations` stream HttpError: `404 - User Not Found`

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving:
- https://github.com/airbytehq/oncall/issues/4648
- https://github.com/airbytehq/oncall/issues/4514

## How
- downgraded the `conversations` stream to use the `2.9` API version, instead of the `2.10`, where the 404 - User Not Found bug occurs for some Customers, that have the `external users` attached to their conversations. 

There are no diffs between `2.9` and `2.10` in terms of the `schema`, so we're safe to do this change.

### CODE REVIEW

**Comment by bazarnov** [comment]
The `pre-release` version of the source-intercom has been published. The tag: `0.6.2-dev.bbd49c0fc2`

**Comment by bazarnov** [comment]
The `pre-release` version of the `source-intercom` has been published. The tag: `0.6.2-dev.dc00d8a8a6`, this one is based on the `Intercom-Version: 2.9` instead of the `2.10`, which appears to have bugs with the `search` method for the `conversations` endpoint.

**Review by askarpets — APPROVED** [review]

**Review by darynaishchenko — APPROVED** [review]

**Comment by bazarnov** [comment]
TODO:
Fix incremental syncs for semi-incremental streams

File: airbyte-integrations/connectors/destination-duckdb/destination_duckdb/destination.py [hunk 1]
```diff
@@ -14,14 +14,7 @@
 import duckdb
```
Comment (bazarnov) [inline airbyte-integrations/connectors/destination-duckdb/destination_duckdb/destination.py:14]: This is not related `badly formatted` file, I've just included it to fix the `master`.

**Comment by bazarnov** [comment]
There are some non-related `formatting` issues observed, will wait until these are fixed. 

https://github.com/airbytehq/airbyte/actions/runs/8396167842/job/22996972694?pr=36277
